src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java

```diff
@@ -206,6 +205,27 @@ private static DetailAST getWhitespaceFollowedNode(DetailAST ast) {
         return whitespaceFollowedAst;
     }
 
+    /**
+     * Returns whether whitespace after a visited node can be ignored. For example, whitespace is
+     * not allowed between a type and an array declarator, except when there is an annotation in
+     * between the type and array declarator.
+     * @param ast the visited node
+     * @return true if the check should ignore whitespace after the ast
+     */
+    private static boolean ignoreWhitespaceAfter(DetailAST ast) {
+        boolean ignoreWhitespace = false;
+        if (ast.getNextSibling() != null) {
+            final DetailAST sibling = ast.getNextSibling();
+            if (sibling.getType() == TokenTypes.ANNOTATIONS) {
+                ignoreWhitespace = true;
+            }
+            else if (sibling.getType() == TokenTypes.ARRAY_DECLARATOR) {
+                ignoreWhitespace = sibling.getFirstChild().getType() == TokenTypes.ANNOTATION;
```

Comment:
tentatively added this case to ignore whitespace in expressions like `new String @B [3];`